src/osd/OSD.cc

```diff
@@ -8061,6 +8063,16 @@ void OSD::handle_op(OpRequestRef& op, OSDMapRef& osdmap)
       return;
     }
 
+    // pool is full ?
+    pg_t _pgid = m->get_pg();
+    int64_t poolid = _pgid.pool();
```

Comment:
pgid and pool id are used pretty soon after this block - you can remove the 2nd lookup below